### PR TITLE
Alt react post

### PR DESCRIPTION
**Pull request by bryanbraun**
You can get [a markdown view here](https://github.com/bryanbraun/bryanbraun.github.io/blob/fe6fb540a0a40691a98eca3835b5b9a85d93bd03/_posts/2019-11-4-web-dev-nirvana-and-why-I-needed-to-let-go-of-reactjs-to-reach-it.md). The images are broken in this view, so you'd have to view them in the "Files Changed".

Or you could run the site locally, with `jekyll serve`, though it'll take a moment to get going since my build is slow 🐌.

### CODE REVIEW

File: _posts/2019-11-4-web-dev-nirvana-and-why-I-needed-to-let-go-of-reactjs-to-reach-it.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+_"Don't we need a bundler to use 3rd-party libraries?"_
+
+Historically, most NPM modules have published their code in a CommonJS format, which is usable by bundlers, but not by browsers. With ES6 modules growing in popularity, this is starting to change. Lots of 3rd-party libraries are available in ES6 module format, and for those that aren't, you have a few options:
+
+- **Import the global version of the package.** This can be loaded in a script tag or [an empty import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Import_a_module_for_its_side_effects_only)).
+- **Use some tooling to do a one-time package conversion.** Pika can [turn ESM packages with CommonJS internal dependencies into web-friendly imports](https://www.pika.dev/blog/pika-web-a-future-without-webpack/). Packages that don't support ESM at all could probably be converted with [the right Rollup plugins](https://github.com/rollup/rollup-plugin-commonjs).
```
Comment (iAmNathanJ) [inline _posts/2019-11-4-web-dev-nirvana-and-why-I-needed-to-let-go-of-reactjs-to-reach-it.md]: This is a nice suggestion. I wouldn't mind seeing a 4th option of maintaining a production only build that does simple minification and conversion of packages that don't come with a module entry. Could also use that as a place to link to the module nomodule pattern. That said, I think it's ok if you leave this as is.

File: _posts/2019-11-4-web-dev-nirvana-and-why-I-needed-to-let-go-of-reactjs-to-reach-it.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+This was *my* app. If this thing needed maintenance, I'd be doing that on my precious nights and weekends. I don't have the time or motivation for that. I desperately needed to build an app with [no nonsense]({{site.url}}/2019/04/16/my-number-one-requirement-for-choosing-a-technology/).
+
+Unfortunately, modern JavaScript development is full of nonsense, and the React ecosystem is no exception.
+
+You can't just create a React app... you have to create an epic toolchain to get that app running in a web browser. If anything breaks in all those layers of npm, webpack, babel, and everything in between, you have to figure out how it all works before you can fix it. Things like [Parcel](https://parceljs.org/) and [create-react-app](https://github.com/facebook/create-react-app) can help, but you're still downloading 200MB of dependencies and hoping [the abstraction doesn't leak](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/). I've done this long enough to know better.
```
[the remark this answers is not in the document]
Comment (bryanbraun) [reply _posts/2019-11-4-web-dev-nirvana-and-why-I-needed-to-let-go-of-reactjs-to-reach-it.md]: Yeah, I mostly wanted to defend against people being like "If you just use `create-react-app` then you don't have to worry about configuring webpack, babel, etc."

I know the up-and-running workflow for create-react-app is pretty good out of the box, so I think I can reword this to be less "hating on CRA" and more "there are hidden costs."

File: _posts/2019-11-4-web-dev-nirvana-and-why-I-needed-to-let-go-of-reactjs-to-reach-it.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+Well, we'd get rid of Babel, since transpiling is a big workflow burden (especially for debugging). Webpack too, since it's inscrutable. Honestly, the fewer dependencies the better, to reduce our upgrade and security burden.
+
+Some people [have experimented](https://medium.com/@chrislewisdev/react-without-npm-babel-or-webpack-1e9a6049714) with these kinds of setups, but they're pretty far off the React happy-path. Specifically, React without JSX is so awkward, that it hardly seems worth it.
+
+So we're stuck, right?
+
+No!
+
+It turns out that the best parts of React aren't in the framework itself... it's in the architecture it embraces. Things like modular self-contained components and explicit state management with a predictable data flow. Reactive UIs.
```
[the remark this answers is not in the document]
Comment (bryanbraun) [reply _posts/2019-11-4-web-dev-nirvana-and-why-I-needed-to-let-go-of-reactjs-to-reach-it.md]: Woah, I didn't know about htm. I'll give it a mention.

**Comment by bryanbraun** [comment]
Made a bunch of update. Still doesn't feel perfect but I'm gonna publish it anyway. 🙈
